nnfabrik/templates/transfer/transferred_trained_model.py

```diff
@@ -0,0 +1,310 @@
+                    # if it exists add that entry to the list of recipes (relevant for a specific transfer step)
+                    transfer_recipe.append(tr & f"transfer_step = {transfer_step}")
+            if not transfer_recipe:
+                return self.proj() - self  # return something empty
+            # join all the recipes (and their post_restr)
+            joined = transfer_recipe[0]
+            if len(transfer_recipe) > 1:
+                for t in transfer_recipe[1:]:
+                    joined *= t  # all combination of recipes
+                joined.post_restr = dj.AndList([recipe.post_restr for recipe in self.transfer_recipe])
+            return joined
+        else:
+            return self.transfer_recipe
+
+    @property
+    def key_source(self):
+        key_source = None
+        if hasattr(self, "transfer_recipe"):  # expand current entries to follow transfer recipes
+            # project (rename) attributes of the existing transferredmodel table to the same name but with prefix "prev"
+            prev_transferred_model = self.proj(
+                prev_model_fn="model_fn",
+                prev_model_hash="model_hash",
+                prev_dataset_fn="dataset_fn",
+                prev_dataset_hash="dataset_hash",
+                prev_trainer_fn="trainer_fn",
+                prev_trainer_hash="trainer_hash",
+                prev_seed="seed",
+                transfer_step="transfer_step + 1",
+                prev_collapsed_history="collapsed_history",
+                _data_transfer="data_transfer",  # rename so this entry in the recipe is not used for restriction
+                prev_step="transfer_step",  # rename so this entry in the recipe is not used for restriction
+            ) * dj.U(
+                "transfer_step",  # make these attributes primary keys
+                "prev_model_fn",
+                "prev_model_hash",
+                "prev_dataset_fn",
+                "prev_dataset_hash",
+                "prev_trainer_fn",
+                "prev_trainer_hash",
+                "prev_seed",
+                "prev_collapsed_history",
+            )
+
+            # get the current transfer step
+            max_transfer_step = prev_transferred_model.fetch("transfer_step").max() if prev_transferred_model else 0
+
+            for transfer_step in range(1, max_transfer_step + 1):
+                # get the transfer recipe
+                recipe = self._transfer_recipe(transfer_step)
+                post_restr = recipe.post_restr if recipe else dj.AndList([])
+
+                # apply the recipe
+                transfer_from = prev_transferred_model * recipe
+                transfers = (
+                    self.model_table * self.dataset_table * self.trainer_table * self.seed_table * transfer_from
+                )  # combine recipe restriction with all possible training combinations
+
+                transfers = transfers * self.CollapsedHistory().proj(
```

Comment:
upon changing the definition of the `CollapsedHistory`, would need to add renaming of

```python
prev_collapsed_history = collapsed_history,
collapsed_history = next_collapsed_history,
```